aws_lambda_powertools/event_handler/openapi/types.py

```diff
@@ -16,13 +16,6 @@
 COMPONENT_REF_TEMPLATE = "#/components/schemas/{model}"
 METHODS_WITH_BODY = {"GET", "HEAD", "POST", "PUT", "DELETE", "PATCH"}
 
-try:
-    from pydantic.version import VERSION as PYDANTIC_VERSION
-
-    PYDANTIC_V2 = PYDANTIC_VERSION.startswith("2.")
-except ImportError:
-    PYDANTIC_V2 = False
-
```

Comment:
for my own understanding... the reason this "lazy" works is that:

* Event Handler entry point is importing `openapi/types.py`
* We were importing Pydantic to confirm it was present at the global scope
* This change moves the try/catch to a separate module which is not requested by `openapi/types.py`
* When `enable_validation=True` is used, we effectively import the rest which will import the new module that imports Pydantic to confirm is installed or not

<img width="737" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/96891182-61a4-45bf-b96b-dca6c3da2d3f">


<img width="973" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/474e5cd7-d413-49ba-a891-992ec1340014">